app/src/main/java/fr/free/nrw/commons/media/ZoomableActivity.kt

```diff
@@ -0,0 +1,527 @@
+                    val imageModifiedSHA1 = CustomSelectorUtils.generateModifiedSHA1(
+                        images!![i],
+                        defaultDispatcher,
+                        this@ZoomableActivity,
+                        fileProcessor,
+                        fileUtilsWrapper
+                    )
+                    isNonActionable = uploadedStatusDao.findByModifiedImageSHA1(
+                        imageModifiedSHA1,
+                        true
+                    )
+                    if (isNonActionable <= 0) {
+                        return i
+                    } else {
+                        continue
+                    }
+                } else {
+                    continue
+                }
+            } else {
+                continue
+            }
+        }
+        return previousPosition
+    }
+
+    /**
+     * Unselect item UI
+     */
+    private fun itemUnselected() {
+        selectedCount!!.visibility = View.INVISIBLE
+    }
+
+    /**
+     * Select item UI
+     */
+    private fun itemSelected(i: Int) {
+        selectedCount!!.visibility = View.VISIBLE
+        selectedCount!!.text = i.toString()
+    }
+
+    /**
+     * Get position of an image from list
+     */
+    private fun getImagePosition(list: ArrayList<Image>?, image: Image): Int {
+        return list!!.indexOf(image)
+    }
+
+    /**
+     * Two types of loading indicators have been added to the zoom activity:
+     * 1.  An Indeterminate spinner for showing the time lapsed between dispatch of the image request
+     * and starting to receiving the image.
+     * 2.  ProgressBarDrawable that reflects how much image has been downloaded
+     */
+    private val loadingListener: ControllerListener<ImageInfo?> =
+        object : BaseControllerListener<ImageInfo?>() {
+            override fun onSubmit(id: String, callerContext: Any) {
+                // Sometimes the spinner doesn't appear when rapidly switching between images, this fixes that
+                spinner!!.visibility = View.VISIBLE
+            }
+
+            override fun onIntermediateImageSet(id: String, imageInfo: ImageInfo?) {
+                spinner!!.visibility = View.GONE
+            }
+
+            override fun onFinalImageSet(
+                id: String,
+                imageInfo: ImageInfo?,
+                animatable: Animatable?
+            ) {
+                spinner!!.visibility = View.GONE
+            }
+        }
+
+    private fun init(imageUri: Uri?) {
+        if (imageUri != null) {
+            val hierarchy = GenericDraweeHierarchyBuilder.newInstance(resources)
+                .setActualImageScaleType(ScalingUtils.ScaleType.FIT_CENTER)
+                .setProgressBarImage(ProgressBarDrawable())
+                .setProgressBarImageScaleType(ScalingUtils.ScaleType.FIT_CENTER)
+                .build()
+            photo!!.hierarchy = hierarchy
+            photo!!.setAllowTouchInterceptionWhileZoomed(true)
```

Comment:
Shouldn't this be `false` ([doc](https://github.com/facebook/fresco/blob/main/samples/zoomable/src/main/java/com/facebook/samples/zoomable/ZoomableDraweeView.java#L190))? That alone does not seem to solve the bug, though, I just tried.